Update README, add umbrella configuration section (#685)
sorentwo/oban: README.md

  priority. Within a particular priority jobs are executed in their scheduled
  order.

### Umbrella Apps

If you need to run Oban from an umbrella application where more than one of

> the child apps need to interact with Oban, you may need to set the :name for

Reviewer: ```suggestion
the child apps need to interact with Oban, you may need to set the `:name` for
```

each child application that configures Oban.

> #### For example
> 
> `MyAppA` may be the app responsible for inserting Oban Jobs, but never
> processing those jobs.
> 
> Configure Oban in `MyAppA`

Reviewer: ```suggestion
For example, your umbrella contains two apps: `MyAppA` and `MyAppB`. `MyAppA` is
responsible for inserting jobs, while only `MyAppB` actually runs any queues.

Configure Oban with a custom name for `MyAppA`:

```

>   ```elixir
>   config :my_app_a, Oban,
>     name: MyAppA.Oban,
>     repo: MyApp.Repo,
>     plugins: [],
>     queues: []
>   ```

Reviewer: ```suggestion
```elixir
config :my_app_a, Oban,
  name: MyAppA.Oban,
  repo: MyApp.Repo
```
```

> `MyAppB` may be the app resonsible for processing Oban jobs.

Reviewer: ```suggestion
Then configure Oban for `MyAppB` with a different name:
```

> Configure Oban in `MyAppB`
>   ```elixir
>   config :my_app_b, Oban,
>     name: MyAppB.Oban,
>     repo: MyApp.Repo,
>     plugins: [...],
>     queues: [:default, :queue_two, :etc]
>   ```

Reviewer: ```suggestion
```elixir
config :my_app_b, Oban,
  name: MyAppB.Oban,
  repo: MyApp.Repo,
  queues: [default: 10]
```
```

> Then when scheduling a job use the configured name to reference the named
> Oban process.
> 
> See:
>   - `insert/2`
>   - `insert/4`
>   - `insert!/2`
>   - `insert_all/2`
>   - `insert_all/4`
> 
>   ```elixir
>   Oban.insert!(MyAppA.Oban, %Oban.Job{})
>   Oban.insert(MyAppA.Oban, %Oban.Job{})
>   Oban.insert_all(MyAppB.Oban, multi, :multiname, [%Oban.Job{}])
>   ```
> 
> For testing the named Oban process, use `drain_queue/2`
> 
>   ```elixir
>   Oban.drain_queue(MyAppB.Oban, queue: :queue_two)
>   ```

Reviewer: ```suggestion
Now, use the configured name when calling functions like `Oban.insert/2`,
`Oban.insert_all/2`, `Oban.drain_queue/2`, etc., to reference the correct Oban
process for the current application.

```elixir
Oban.insert(MyAppA.Oban, MyWorker.new(%{}))
Oban.insert_all(MyAppB.Oban, multi, :multiname, [MyWorker.new(%{})])
Oban.drain_queue(MyAppB.Oban, queue: :default)
```
```
